content/posts/2021/07-15-salt-nornir-intro.md

```diff
@@ -0,0 +1,229 @@
+    |                              |
+    +------------------------------+
+```
+Plugins can be developed independently of the main Core package and registered during runtime to execute tasks on devices.
+
+## SALTSTACK intro
+
+Key aspects of SALT architecture could be summarized as below:
+
+```
+                      |-RUNNERS    |-REACTOR      |-SCHEDULERS
+                      |-CLOUD      |-FILE SERVER  |-STATE 
+                      |-WHEEL      |-AUTH         |-RENDERERS 
+                      |-MINE       |-PILLAR       |-OUTPUT 
+                      +------------+--------------+               
+                                   |                     
+                    |           MASTER             |
+                    |                              |
+                    +----------Server/VM-----------+
+    ENGINES                        |                       API
+       |                           |                        |
+       |                           |                        |
+===========================ZeroMQ  EVENT BUS=============================
+                 |                                    |
+                 |                                    |
+      +------Server/VM-------+              +------Server/VM-------+
+      |                      |              |                      |
+      |  OS (Windows/Linux)  |              |  OS (Windows/Linux)  |
+      |   |-MINION process   |              |   |-PROXY-MINION     |   
+      |     |-BEACON         |              |     |-BEACON         |   
+      |     |-GRAINS         |              |     |-GRAINS         |   
+      |     |-RETURNER       |              |     |-RETURNER       |   
+      |     |-SCHEDULER      |              |     |-SCHEDULER      |   
+      |     |-EXECUTION      |              |     |-EXECUTION      |
+      |       MODULES        |              |       MODULES        |
+      |                      |              |                      |
+      +----------------------+              +----------------------+
+                                                      |         
+                                                   SSH/API    
+                                                      |         
+                                                   +------+     
+                                                   |DEVICE|     
+                                                   +------+ 
+```
+SALT Master is the HUB of the overall system, it communicates with Minions to execute various tasks. SALTSTACK uses [modules](https://docs.saltproject.io/en/latest/ref/index.html) to address various use cases.
+
+In the simplest (master-less) case, no Master required and only Minion process need to run on the system. But (probably) most common deployment is a two tier hierarchy where Master controls (many) Minions. Each Minion acts as an agent natively running on the Operating System of the system being managed. 
```

Comment:
"no Master IS required and only THE/A Minion process is needed on the system"

"THE most common"

"where THE Master"

I'd also remove the parenthesis. The words inside them are ok